src/lib/gateway/gateway-request.ts

```diff
@@ -16,37 +16,43 @@ type GatewayRequest =
   | ConditionalTransferRequest;
 
 interface DepositRequest {
+  txId?: number;
```

Comment:
This member (and sometimes others) repeats itself.
Why not have another interface for interface to a request for a transaction, or simple a transaction?
Something like:
```
interface TransactionRequest {
  txId? : number;
}
```
and then:
```
interface DepositRequest extends TransactionRequest {
  vaultId :  number;
  ... 
}
```